Open Judge System/Services/OJS.Services/Cache/CacheItemsProviderService.cs

```diff
@@ -131,5 +136,53 @@ void RemoveCacheFromCategory(int contestCategoryId)
                 this.cache.Remove(parentCategoriesCacheId);
             }
         }
+
+        public IEnumerable<HomeContestViewModel> GetActiveContests()
+        {
+            var upcomingMaxTime = DateTime.Now.AddHours(2);
+
+            var cachedResult = this.cache.Get(
+                CacheConstants.ActiveContests,
+                () => this.contestsData
+                    .GetAllUpcoming()
+                    .Where(x => x.StartTime.HasValue && x.StartTime <= upcomingMaxTime)
+                    .Select(HomeContestViewModel.FromContest)
+                    .Concat(
+                        this.contestsData.GetAllCompetable()
+                            .OrderBy(ac => ac.EndTime)
```

Comment:
Shouldn't this `OrderBy` should be on the whole collection, not only on `Competable`